### PR TITLE
disttask: add available resource param to step executor

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #49008

Problem Summary:

### What changed and how does it work?
- add available resource param to runSubtask. each business logic should not use resource over this(will be added later)
- some renaming
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/taskexecutor/task_executor.go [hunk 1]
```diff
@@ -147,9 +147,8 @@ func (*BaseTaskExecutor) Init(_ context.Context) error {
 	return nil
 }
 
-// Run start to fetch and run all subtasks of the task on the node.
-// TODO: rename it too when we move the loop here from manager.
-func (s *BaseTaskExecutor) Run(ctx context.Context, task *proto.Task) (err error) {
+// RunStep start to fetch and run all subtasks of the task on the node.
```
Comment (ywqzzy) [inline pkg/disttask/framework/taskexecutor/task_executor.go:150]:
```suggestion
// RunStep start to fetch and run all subtasks for the step of task on the node.
```

File: pkg/disttask/framework/taskexecutor/manager.go [hunk 2]
```diff
@@ -410,6 +434,14 @@ func (m *Manager) handleExecutableTask(task *proto.Task) {
 	}
 }
 
+func (m *Manager) getStepResource(concurrency int) *proto.StepResource {
+	return &proto.StepResource{
+		CPU: proto.NewAllocatable(int64(concurrency)),
+		// same proportion as CPU
+		Mem: proto.NewAllocatable(int64(float64(concurrency) / float64(m.totalCPU) * float64(m.totalMem))),
```
Comment (ywqzzy) [inline pkg/disttask/framework/taskexecutor/manager.go:441]: assert concurrency < totalCPU?

Comment (D3Hunter) [reply pkg/disttask/framework/taskexecutor/manager.go:441]: it will fail to alloc slots, shouldn't reach here

Comment (ywqzzy) [reply pkg/disttask/framework/taskexecutor/manager.go:441]: Yep, I mean just add some comments

**Review by ywqzzy — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**